File: 141-linked-list-cycle/141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        if(!head || !(head->next)){
            return false;
        }
        ListNode *first = head,*second = head->next;
        while(second && second->next && first != second){
            first = first->next;
            second = second->next->next;
        }
        return first == second;
    }
};

File: 141-linked-list-cycle/141-linked-list-cycle.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
    if head == nil || head.Next == nil {
        return false
    }
    var first,second *ListNode
    for first,second = head,(head.Next);second != nil && second.Next != nil && first != second; first,second = first.Next,second.Next.Next{

    }
    return first == second
}